api/hypershift/v1beta1/nodepool_types.go

```diff
@@ -359,26 +359,33 @@ type InPlaceUpgrade struct {
 // NodePoolManagement specifies behavior for managing nodes in a NodePool, such
 // as upgrade strategies and auto-repair behaviors.
 type NodePoolManagement struct {
-	// UpgradeType specifies the type of strategy for handling upgrades.
+	// upgradeType specifies the type of strategy for handling upgrades.
+	// This can be either "Replace" or "InPlace".
+	// "Replace" will update Nodes by recreating the underlying instances.
+	// "InPlace" will update Nodes by applying changes to the existing instances.
```

Comment:
This causes the nodes to reboot right? Should we also mention that here?

Reply:
>This causes the nodes to reboot right

Not necessarily, I'll mention it.